hubblestack/utils/signing.py

```diff
@@ -229,34 +307,44 @@ def __init__(self, public_crt, ca_crt):
             try:
                 ossl.X509StoreContext(self.store, i).verify_certificate()
                 status = STATUS.VERIFIED
                 self.trusted.append(digest)
-                log.debug('  marking verified')
+                if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
+                    if status == STATUS.FAIL: log_level = log.critical
+                    elif status == STATUS.UNKNOWN: log_level = log.error
+                    else: log_level = log.splunk
+                str_public = stringify_cert_files(public_crt)
+                msg = 'public cert | file: "{}" | status : "{}" | digest: "{}"'
+                log_level(msg.format(str_public, status, digest))
             except ossl.X509StoreContextError as exception_object:
                 code, depth, message = exception_object.args[0]
-                log.debug('authentication of %s failed: code=%s depth=%s, message=%s',
-                    digest, code, depth, message)
-                # from openssl/x509_vfy.h
-                # define X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT         2
-                # define X509_V_ERR_UNABLE_TO_GET_CRL                 3
-                # define X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY 20
-                # define X509_V_ERR_CERT_UNTRUSTED                    27
-                # define X509_V_ERR_UNABLE_TO_GET_CRL_ISSUER          33
                 if code in (2,3,20,27,33):
-                    # we just don't have the required info, it's not failing to
-                    # verify not exactly, but it's definitely not verified
-                    # either
-                    log.debug(' code=%d is alright-ish though. setting status to UNKNOWN', code)
+                    # from openssl/x509_vfy.h or 
+                    # https://www.openssl.org/docs/man1.1.0/man3/X509_STORE_CTX_set_current_cert.html
+                    # X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT         2
+                    # X509_V_ERR_UNABLE_TO_GET_CRL                 3
+                    # X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY 20
+                    # X509_V_ERR_CERT_UNTRUSTED                    27
+                    # X509_V_ERR_UNABLE_TO_GET_CRL_ISSUER          33
                     status = STATUS.UNKNOWN
-                # define X509_V_ERR_CERT_HAS_EXPIRED                  10
-                # define X509_V_ERR_CRL_HAS_EXPIRED                   12
-                # XX # if code in (10,12):
-                # XX #     return # .... is this even the right idea? should we do this through conext flags?
+                if code in (10,12):
+                    # X509_V_ERR_CERT_HAS_EXPIRED                  10
+                    # X509_V_ERR_CRL_HAS_EXPIRED                   12
+                    status = STATUS.FAIL
+                # log at either log.error or log.critical according to the error code
+                if check_verif_timestamp(digest, dampener_limit=seconds_day) == True:
```

Comment:
Redundant comparison to True, check_verif_timestamp() returns bool value already.